### PR TITLE
feat: Add Pod Scheduling SLI Duration metric

### PR DESCRIPTION
**Pull request by helayoty**
Add a new scheduler metric, `pod_scheduling_sli_duration_seconds`.

#### What type of PR is this?

/kind feature
/sig scheduling

#### What this PR does / why we need it:
As a final change to the update related to KEP [#3521](https://github.com/kubernetes/enhancements/pull/4065), discussion can be found [here](https://github.com/kubernetes/enhancements/pull/4065#discussion_r1231487092).

#### Which issue(s) this PR fixes:
NA

#### Does this PR introduce a user-facing change?
Yes. New metric, `pod_scheduling_sli_duration_seconds`

```release-note
Add a new scheduler metric, `pod_scheduling_sli_duration_seconds`, and start the deprecation for `pod_scheduling_duration_seconds`.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

KEP [#3521](https://github.com/kubernetes/enhancements/pull/4065)

```docs

```

### CODE REVIEW

File: pkg/scheduler/metrics/metrics.go
Comment (Huang-Wei) [inline pkg/scheduler/metrics/metrics.go:117]:
```suggestion
	// PodSchedulingDuration is deprecated as of Kubernetes v1.28, and will be removed
	// in v1.31. Please use PodSchedulingSLIDuration instead.
```

Comment (helayoty) [reply pkg/scheduler/metrics/metrics.go]: Addressed.

**Review by Huang-Wei — COMMENTED** [review]
LGTM on scheduling side.

/approve

File: pkg/scheduler/metrics/metrics.go
Comment (alculquicondor) [inline pkg/scheduler/metrics/metrics.go:135]:
```suggestion
			Help:      "E2e latency for a pod being scheduled, from the time the pod enters the scheduling queue an d might involve multiple scheduling attempts.",
```

File: pkg/scheduler/metrics/metrics.go
Comment (helayoty) [reply pkg/scheduler/metrics/metrics.go]: Addressed. Thanks for the hint

**Comment by valaparthvi** [comment]
Hello @helayoty,
I am checking in on behalf of the Bug Triage release team. We are only a week away from Code Freeze ([01:00 UTC Wednesday 19th July 2023 / 18:00 PDT Tuesday 18th July 2023](https://everytimezone.com/s/72ee8496)). Please ensure your PR goes in by then if you are targeting 1.28 release. 

Please let me know if I can be of any assistance :)

**Comment by Huang-Wei** [comment]
LGTM. @helayoty could you squash the commits?

File: pkg/scheduler/metrics/metrics.go
Comment (alculquicondor) [inline pkg/scheduler/metrics/metrics.go:135]:
```suggestion
			Help:      "E2e latency for a pod being scheduled, from the time the pod enters the scheduling queue and might involve multiple scheduling attempts.",
```

**Review by alculquicondor — COMMENTED** [review]
/label tide/merge-method-squash

File: pkg/scheduler/metrics/metrics.go
Comment (logicalhan) [inline pkg/scheduler/metrics/metrics.go:126]: Was this already stable?

Comment (alculquicondor) [reply pkg/scheduler/metrics/metrics.go:126]: yes, it is.

**Comment by furkatgofurov7** [comment]
@helayoty hi, Bug Triage Lead here 👋🏼

Since it was communicated in earlier weeks (3 & 1 weeks before), Code Freeze Deadline passed and, this PR has not made to the current milestone, hence I am moving this PR to the next milestone.

/milestone next-candidate

**Comment by helayoty** [comment]
@furkatgofurov7  @alculquicondor  , just back from vacation. The PR was ready to merge but just needed a squash. I did that now, and it's up to you to decide if we need an exception or if it can wait for the next release.

**Review by logicalhan — APPROVED** [review]
/lgtm

**Comment by furkatgofurov7** [comment]
> @furkatgofurov7 @alculquicondor , just back from vacation. The PR was ready to merge but just needed a squash. I did that now, and it's up to you to decide if we need an exception or if it can wait for the next release.

@helayoty I am fine with both options, although I do not know if we have still time for exception. Tagging Release Lead for visibilty

/cc @gracenng

**Comment by alculquicondor** [comment]
I'm inclined to leave this for the next release

**Pull request merged**
